src/Parsers/PRQL/ParserPRQLQuery.cpp

```diff
@@ -0,0 +1,84 @@
+#include <string>
+#include <Parsers/PRQL/ParserPRQLQuery.h>
+
+#include "Parsers/Lexer.h"
+#include "config.h"
+
+#if USE_PRQL
+#    include <prql.h>
+#endif
+
+#include <Parsers/ParserQuery.h>
+#include <Parsers/ParserSetQuery.h>
+#include <Parsers/parseQuery.h>
+#include <base/scope_guard.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int SYNTAX_ERROR;
+    extern const int SUPPORT_IS_DISABLED;
+}
+
+bool ParserPRQLQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ParserSetQuery set_p;
+
+    if (set_p.parse(pos, node, expected))
```

Comment:
Interesting, this is to make it able to switch back to another parser, right?
Maybe it worth to put a comment so that reader will not be surprised.

BTW other parsers (i.e. KQL) first parse with their parser and only if it fails with the `ParserSetQuery`

Reply:
No, you are right, let's keep it this way, and maybe later it worth to update the other parser as well, since KQL behaves differently.